content/en/serverless/step_functions/merge-step-functions-lambda.md

```diff
@@ -178,9 +178,49 @@ To link your Step Function traces to nested Step Function traces, configure your
 }
 {{< /highlight >}}
 
-## Merge Lambda -> Step Functions -> Lambda
+## Merge Any Upstream Service -> Step Functions -> Lambda
 
-This capability is not supported.
+### Requirements
+Node.js (layer v113+) or Python (layer v103+) runtimes.
+
+Your State Machine Definition must be using `JSONata` as the query language. This can be enabled by setting `"QueryLanguage": "JSONata"` at the top-level of the State Machine Definition.
+
+### Setup
+
+On the Lambda Task, set the `Arguments` key as follows: 
+
+```json
+"Arguments": {
+  "Payload": "{% ($execInput := $states.context.Execution.Input; $hasDatadogTraceId := $exists($execInput._datadog.`x-datadog-trace-id`); $hasDatadogRootExecutionId := $exists($execInput._datadog.RootExecutionId); $ddTraceContext := $hasDatadogTraceId ? {'x-datadog-trace-id': $execInput._datadog.`x-datadog-trace-id`, 'x-datadog-tags': $execInput._datadog.`x-datadog-tags`} : {'RootExecutionId': $hasDatadogRootExecutionId ? $execInput._datadog.RootExecutionId : $states.context.Execution.Id}; $sfnContext := $merge([$states.context, {'Execution': $sift($states.context.Execution, function($v, $k) { $k != 'Input' })}]); $merge([$states.input, {'_datadog': $merge([$sfnContext, $ddTraceContext, {'serverless-version': 'v1'}])}])) %}"
+  ...
+}
+```
+
+The `JSONata` expression will do the following:
```

Comment:
The section above had an explanation for the `JsonMerge` expression so I thought it made sense to have some explanation here

If this is too much detail, I think we could also say
> The `JSONata` expression will merge the upstream service's context with the current Step Functions context object and the Lambda state's input.